pkg/skaffold/config/config.go

```diff
@@ -17,117 +17,9 @@ limitations under the License.
-type BuildType struct {
-	LocalBuild       *LocalBuild       `yaml:"local"`
-	GoogleCloudBuild *GoogleCloudBuild `yaml:"googleCloudBuild"`
-}
-
-// LocalBuild contains the fields needed to do a build on the local docker daemon
-// and optionally push to a repository.
-type LocalBuild struct {
-	SkipPush *bool `yaml:"skipPush"`
-}
-
-type GoogleCloudBuild struct {
-	ProjectID string `yaml:"projectId"`
-}
-
-// DeployConfig contains all the configuration needed by the deploy steps
-type DeployConfig struct {
-	Name       string `yaml:"name"`
-	DeployType `yaml:",inline"`
-}
-
-// DeployType contains the specific implementation and parameters needed
-// for the deploy step. Only one field should be populated.
-type DeployType struct {
-	HelmDeploy    *HelmDeploy    `yaml:"helm"`
-	KubectlDeploy *KubectlDeploy `yaml:"kubectl"`
-}
-
-// KubectlDeploy contains the configuration needed for deploying with `kubectl apply`
-type KubectlDeploy struct {
-	Manifests []Manifest `yaml:"manifests"`
-}
-
-type Manifest struct {
-	Paths      []string          `yaml:"paths"`
-	Parameters map[string]string `yaml:"parameters"`
-}
-
-type HelmDeploy struct {
-	Releases []HelmRelease `yaml:"releases"`
-}
-
-type HelmRelease struct {
-	Name           string            `yaml:"name"`
-	ChartPath      string            `yaml:"chartPath"`
-	ValuesFilePath string            `yaml:"valuesFilePath"`
-	Values         map[string]string `yaml:"values"`
-	Namespace      string            `yaml:"namespace"`
-	Version        string            `yaml:"version"`
-}
-
-// Artifact represents items that need should be built, along with the context in which
-// they should be built.
-type Artifact struct {
-	ImageName      string             `yaml:"imageName"`
-	DockerfilePath string             `yaml:"dockerfilePath"`
-	Workspace      string             `yaml:"workspace"`
-	BuildArgs      map[string]*string `yaml:"buildArgs"`
-}
-
-// DefaultDevSkaffoldConfig is a partial set of defaults for the SkaffoldConfig
-// when dev mode is specified.
-// Each API is responsible for setting its own defaults that are not top level.
-var DefaultDevSkaffoldConfig = &SkaffoldConfig{
-	Build: BuildConfig{
-		TagPolicy: constants.DefaultDevTagStrategy,
-	},
-}
-
-// DefaultRunSkaffoldConfig is a partial set of defaults for the SkaffoldConfig
-// when run mode is specified.
-// Each API is responsible for setting its own defaults that are not top level.
-var DefaultRunSkaffoldConfig = &SkaffoldConfig{
-	Build: BuildConfig{
-		TagPolicy: constants.DefaultRunTagStrategy,
-	},
-}
-
-// Parse reads from an io.Reader and unmarshals the result into a SkaffoldConfig.
-// The default config argument provides default values for the config,
-// which can be overridden if present in the config file.
-func Parse(config []byte, defaultConfig *SkaffoldConfig) (*SkaffoldConfig, error) {
-	if err := yaml.Unmarshal(config, defaultConfig); err != nil {
-		return nil, err
-	}
+type SkaffoldConfig = v1alpha1.SkaffoldConfig
```

Comment:
why do we do this instead of just using v1alpha1 everywhere?

Reply:
I wanted it to be more explicit which `SkaffoldConfig` struct is the "latest" one. The components that made up that config are versioned as well, but won't always be the same version (i.e. you could use a `v1alpha1.Artifact` and a `v2.HelmDeploy` to create a new config version). Older versions might still be used throughout the code to create the runner, watcher, etc so those can be imported directly from their version packages.

The other option is to type alias every single config component in this file, something like
```
type Artifact = v1alpha1.Artifact
type HelmDeploy = v2.HelmDeploy
...
```

but the way it is now, if you change the `v1alpha1` version of a config component it will be very explicit what in the existing code broke because of your change. I'm open to hearing arguments for doing it the other way though.